### PR TITLE
Add TileEntity registry and TileEntity types

### PR DESCRIPTION
**Pull request by williambl**
Based off the entity registry code.

~~!!Untested!! works with vanilla. Not tested with custom mod tile entities.~~ Now includes test mod.

### CODE REVIEW

**Comment by valoeghese** [comment]
Test your code thanks

**Comment by valoeghese** [comment]
otherwise make it a draft pr :wink:

**Comment by valoeghese** [comment]
The reason i ask for all PRs to be tested is because of #19, where I assumed it worked since I based it off of one of my mods, but it was broken

**Comment by williambl** [comment]
Yeah, makes sense. I'll mark as draft once I'm at my pc (or i work out how to do it through the app)

**Comment by valoeghese** [comment]
Any progress on this?

**Comment by valoeghese** [comment]
Writing a test mod so we can get this merged as part of 1.0.0 would be cool

**Comment by williambl** [comment]
Sure I'll look into it

**Comment by BoogieMonster1O1** [comment]
any progress on getting this merged?

**Comment by williambl** [comment]
> any progress on getting this merged?

It's ready, just needs reviews

**Review by BoogieMonster1O1 — APPROVED** [review]
pls merge k thx

**Comment by valoeghese** [comment]
Can this be retargeted and made into a `tile-entities` module for 1.0.x?
If you have trouble setting up/testing in the multiproject workspace let me know

**Comment by valoeghese** [comment]
Actually you could probably just add this to the `registries` module tbh

**Comment by valoeghese** [comment]
You can include other modules for testing using `testImplementation` (see how I do it in attached data)

**Comment by valoeghese** [comment]
And sorry for making this PR wait so long

File: src/main/java/io/github/minecraftcursedlegacy/impl/registry/TileEntityTypeRegistry.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	}
+
+	@Override
+	protected void beforeRemap() {
+		AccessorTileEntity.setClassToIdMap(new HashMap<>());
+		AccessorTileEntity.setIdToClassMap(new HashMap<>());
+	}
+
+	@Override
+	protected void onRemap(TileEntityType remappedValue, int newSerialisedId) {
```
Comment (valoeghese) [inline src/main/java/io/github/minecraftcursedlegacy/impl/registry/TileEntityTypeRegistry.java:41]: This seems obselete. TileEntities don't need a registry if they're not stored as an int in vanilla?

**Review by valoeghese — CHANGES_REQUESTED** [review]
If you can do it in vanilla fine and it uses string ids, the registry seems obselete? If the accessor is still necessary for making a tile entity, please provide an API for using the necessary methods from the accessor in a separate module from registries.

File: src/main/java/io/github/minecraftcursedlegacy/impl/registry/TileEntityTypeRegistry.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+		AccessorTileEntity.callRegister(remappedValue.getClazz(), remappedValue.getVanillaRegistryStringId());
```
Comment (valoeghese) [inline src/main/java/io/github/minecraftcursedlegacy/impl/registry/TileEntityTypeRegistry.java:42]: This seems like the only necessary part of the PR to me.

**Comment by williambl** [comment]
Superceded by #31

**Pull request closed without merging**